### PR TITLE
refactor: VTGate executor with a runnable context package

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR refactor the VTGate executor to have `session` and `vcursor` in `executorcontext` package so that planbuilder can use the common `VCursor` implementation in the plan tests.

## Checklist

-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/test/vschemawrapper/vschema_wrapper.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ import (
 var _ plancontext.VSchema = (*VSchemaWrapper)(nil)
 
 type VSchemaWrapper struct {
```
Comment (systay) [inline go/test/vschemawrapper/vschema_wrapper.go:44]: While we are here, we should some welcoming comments explaining the why of this code

File: go/vt/vtgate/executorcontext/vcursor_impl.go [hunk 2]
```diff
@@ -59,38 +60,60 @@ import (
...
-		showTablets(filter *sqlparser.ShowFilter) (*sqltypes.Result, error)
-		showVitessMetadata(ctx context.Context, filter *sqlparser.ShowFilter) (*sqltypes.Result, error)
-		setVitessMetadata(ctx context.Context, name, value string) error
+		ShowVitessReplicationStatus(ctx context.Context, filter *sqlparser.ShowFilter) (*sqltypes.Result, error)
+		ShowShards(ctx context.Context, filter *sqlparser.ShowFilter, destTabletType topodatapb.TabletType) (*sqltypes.Result, error)
+		ShowTablets(filter *sqlparser.ShowFilter) (*sqltypes.Result, error)
+		ShowVitessMetadata(ctx context.Context, filter *sqlparser.ShowFilter) (*sqltypes.Result, error)
+		SetVitessMetadata(ctx context.Context, name, value string) error
 
 		// TODO: remove when resolver is gone
```
Comment (systay) [inline go/vt/vtgate/executorcontext/vcursor_impl.go:109]: can we remove this TODO now? 🥹

File: go/vt/vtgate/executorcontext/vcursor_impl.go [hunk 3]
```diff
@@ -140,23 +160,25 @@ type (
...
+// newVcursorImpl creates a VCursorImpl. Before creating this object, you have to separate out any marginComments that came with
 // the query and supply it here. Trailing comments are typically sent by the application for various reasons,
 // including as identifying markers. So, they have to be added back to all queries that are executed
 // on behalf of the original query.
-func newVCursorImpl(
+func NewVCursorImpl(
 	safeSession *SafeSession,
 	marginComments sqlparser.MarginComments,
-	executor *Executor,
+	executor iExecute,
```
Comment (systay) [inline go/vt/vtgate/executorcontext/vcursor_impl.go:170]: very nice! so much win

File: go/vt/vtgate/executorcontext/vcursor_impl_test.go [hunk 4]
```diff
@@ -397,3 +393,113 @@ func TestRecordMirrorStats(t *testing.T) {
...
+func (f fakeExecutor) AddWarningCount(name string, value int64) {
+	// TODO implement me
+	panic("implement me")
+}
+
+var _ iExecute = (*fakeExecutor)(nil)
+
+type fakeObserver struct{}
+
+func (f fakeObserver) Observe(*sqltypes.Result) {
```
Comment (systay) [inline go/vt/vtgate/executorcontext/vcursor_impl_test.go:502]: the `nullResultsObserver` already exists and does the same thing

**Review by systay — APPROVED** [review]

File: go/vt/vtgate/executorcontext/safe_session.go [hunk 5]
```diff
@@ -410,15 +468,22 @@ func (session *SafeSession) findSessionLocked(keyspace, shard string, tabletType
 	return nil
 }
 
+type myShardActionInfo interface {
```
Comment (GuptaManan100) [inline go/vt/vtgate/executorcontext/safe_session.go:471]: myShardActionInfo doesn't sound like a great name. Why the `my`. Why not just `shardActionInfo`

**Review by GuptaManan100 — APPROVED** [review]
Just a nomenclature thing. Totally your call.

**Pull request merged**